### ISSUE TITLE
Unable to parse heredoc inside a block properly

### ISSUE THREAD

**Issue by bpradeep20** [etingof/apacheconfig#16]
**Config:**
```
<main>
    PYTHON <<MYPYTHON
        def a():
            x = y
            return
    MYPYTHON
</main>
```

With python version of config parser I am getting parse error

But with perl version:

```
$VAR1 = {
          'main' => {
                      'PYTHON' => 'def a():
        x = y
        return'
                    }
        };
```

Can you please check?

**Comment by etingof** [comment]
The issue here is that closing `MYPYTHON` anchor is indented. I've modified the lexer to allow that (see commit 8a9f5b34bb1bd07e44fa94fb0b9b76181d37bb58).

But now I am curious if this is the right thing to do? How would you include `MYPYTHON` into the heretic block? There should be some sort of escaping, right?

**Comment by etingof** [comment]
Hmm, is it a good idea to allow whitespaces around the anchor? At least with bash, this is [not allowed](http://tldp.org/LDP/abs/html/here-docs.html#FTN.AEN17822).

**Comment by bpradeep20** [comment]
This is working with current version of code.

**Comment by etingof** [comment]
It does, but I feel like this may break things elsewhere. May be we should better make this behavior configurable.

**Comment by bpradeep20** [comment]
I agree. It would be great of we have some option to enable PERL heredoc and keep default as BASH here doc.

**Comment by etingof** [comment]
Hey, @sydneyli! Do you think this has been fixed lately?

**Comment by sydneyli** [comment]
The behavior in the original issue fixed, but extra whitespaces after the closing heredoc anchor seems to still break things.
